lib/src/sharding_manager.dart

```diff
@@ -108,9 +108,36 @@ class ShardingManager {
     return recommended;
   }
 
+  Future<int> _getMaxConcurrency() async {
+    if (token == null) {
+      return 1;
+    }
+
+    INyxx client = NyxxFactory.createNyxxRest(token!, GatewayIntents.none, Snowflake.zero());
+
+    await client.connect();
+
+    IHttpResponse gatewayBot = await (client.httpEndpoints as HttpEndpoints).getGatewayBot();
+
+    if (gatewayBot is IHttpResponseError) {
+      throw ShardingError('Cannot connect to Discord to get max connection concurrency: [$gatewayBot]');
+    }
+
+    await client.dispose();
+
+    int maxConcurrency = (gatewayBot as IHttpResponseSucess).jsonBody['session_start_limit']['max_concurrency'] as int;
+
+    _logger.info('Got max concurrency: $maxConcurrency');
```

Comment:
Maybe this could be debugging only? Or have better wording?
```suggestion
    _logger.info('Max concurrency: $maxConcurrency');
```

Reply:
Not for nyxx rest client (rest clients don't have a shard manager), and I'm trying to avoid connecting any shards outside of the spawned processes.

Reply:
If no token is given, this message isn't logged.
If a token is given, similarly to the total shard count, a message should be logged to indicate that a result was fetched from the API.